src/lib/containers/Markdown.js

```diff
@@ -0,0 +1,588 @@
+            return null
+        }
+        // init plot_data
+        let plot_data = [];
+        let orientation = isHorizontal ? "v" : "h";
+        let headers = raw_plot_data.queryResult.queryResults.headers;
+        for (let i = 0; i < headers.length - 1; i++) {
+            // make an entry for each set of data points
+            plot_data[i] = {
+                x: [],
+                y: [],
+                name: headers[i + 1].name,
+                type: type.toLowerCase(),
+                orientation: orientation
+            };
+        }
+        // grab all the data
+        for (let i = 0; i < raw_plot_data.queryResult.queryResults.rows.length; i++) {
+            let row = raw_plot_data.queryResult.queryResults.rows[i];
+            for (let j = 1; j < row.values.length; j++) {
+                // create pairs of data
+                let row_values = row.values;
+                plot_data[j - 1].x.push(row_values[0]);
+                plot_data[j - 1].y.push(row_values[j]);
+            }
+        }
+        // error with clearing html - "" is not a function, wrapping in try/catch prevents the error
+        // although it doesn't catch it.
+        try {
+            elementBundle.element.innerHTML = ""; // clear formatting (e.g. <Synapse Widget></SynapseWidget>)
+        }
+        catch (e) {
+            console.log('element bundle error ', e);
+        }
+        // responsive plot
+        // https://plot.ly/javascript/responsive-fluid-layout/#responsive--fluid-layout
+        (function () {
+            var d3 = window.Plotly.d3;
+            var WIDTH_IN_PERCENT_OF_PARENT = 100, HEIGHT_IN_PERCENT_OF_PARENT = 75;
+            var gd3 = d3.select(elementBundle.element)
+                .append('div')
+                .style({
+                    width: WIDTH_IN_PERCENT_OF_PARENT + '%',
+                    'margin-left': (100 - WIDTH_IN_PERCENT_OF_PARENT) / 5 + '%',
+                    height: HEIGHT_IN_PERCENT_OF_PARENT + 'vh',
+                    'margin-top': (100 - HEIGHT_IN_PERCENT_OF_PARENT) / 5 + 'vh'
+                });
+            var gd = gd3.node();
+            window.Plotly.plot(gd, plot_data, layout, config);
+            window.onresize = function () {
+                window.Plotly.Plots.resize(gd);
+            };
+        })();
+    }
+
+    /**
+     * Get data for plotly
+     *
+     * @param {*} widgetparamsMapped
+     * @returns data corresponding to plotly widget
+     * @memberof Markdown
+     */
+    getPlotlyData(widgetparamsMapped) {
+
+        let raw_plot_data = {}
+
+        // step 1: get init query with maxRowsPerPage calculated
+        let queryRequest = {
+            concreteType: "org.sagebionetworks.repo.model.table.QueryBundleRequest",
+            entityId: this.state.ownerId,
+            query: {
+                isConsistent: false,
+                limit: 150,
+                partMask: 9,  // get query results and max rows per page
```

Comment:
Use constants to verify that you're asking for the minimum amount of work.  
Like: 
partMask: SynapseConstants.BUNDLE_MASK_QUERY_RESULTS 
| SynapseConstants.BUNDLE_MASK_QUERY_SELECT_COLUMNS